src/platform/plugins/shared/controls/public/controls/esql_control/get_esql_control_factory.test.tsx

```diff
@@ -120,6 +127,49 @@ describe('ESQLControlApi', () => {
         expect(mockIsSuccess).toHaveBeenCalledTimes(2);
       });
     });
+
+    test('should load availableOptions but not serialize them', async () => {
```

Comment:
How about moving this test to src/platform/plugins/shared/controls/public/controls/esql_control/esql_control_selections.test.ts since `initializeESQLControlSelections` is responsible for managing this subscription.

Reply:
This is more of an integration test. The component always uses the `availableOptions` state internally to render the list of available options in the UI regardless of whether it's a Values From Query or Static Values control. We only want to serialize and save it for Static Values control types, and only fetch at runtime for Values From Query.

This does make it a bit confusing, but since this is a high-priority fix and we're planning on cleaning up this logic with the Data/ESQL control factory unification, I opted not to refactor here.